nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -1233,6 +1233,65 @@ public void OptionalEmptinessHandlerTest() {
         .doTest();
   }
 
+  @Test
+  public void OptionalEmptinessHandlerWithCustomPathTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:UnannotatedSubPackages=com.uber.lib.unannotated",
+                "-XepOpt:NullAway:CheckOptionalEmptiness=true",
+                "-XepOpt:NullAway:OptionalClassPath=com.google.common.base.Optional"))
```

Comment:
:+1: Would suggest also:

1) Testing for `"-XepOpt:NullAway:OptionalClassPath=\"\""` (I'd say this should just raise a config error if `XepOpt:NullAway:CheckOptionalEmptiness` is set)
2) Testing for `"-XepOpt:NullAway:OptionalClassPath=java.lang.Optional,com.google.common.base.Optional"`

Reply:
Should we not have `java.utils.Optional` by default in the set?

Reply:
The way I would do it, if possible:

1) No option at all passed in the command line => set is (`java.utils.Optional`)
2) Explicit option passed, empty or unparsable => error
3) Explicit option passed, parsable non-empty set => the classes in that set, and nothing else

This allows people using this handler for their own custom `Optional` without enabling it for `java.utils.Optional`. Then again, not sure if it's more consistent with how we do other stuff to just have  `java.utils.Optional` always on (when `CheckOptionalEmptiness` is set), then rename this option to something like `CustomOptionalClasses`. Maybe on second thought, I am liking this later way of doing it better.

@msridhar , thoughts ?

Reply:
I think we should always check `java.util.Optional` and this option should add additional aliases for it.

Reply:
Okay then. So first two cases as Lazaro described and and in case #3 Explicit option passed, parsable non-empty set => the classes in that set + `java.utils.Optional`. 
That works @lazaroclapp?

Reply:
I am not sure if we really need the unparsable check. Since we do not have it for other configs. Also I'm not sure how to define it exactly. So let me know if you think it should be necessary. 
@msridhar @lazaroclapp

Reply:
I think given what Manu is saying, then there is just two cases:

1. No option at all passed in the command line => set is (java.utils.Optional)
2. `CustomOptionalClasses=a.b.c.A1;a.b.c.A2` => set is (java.utils.Optional,a.b.c.A1,a.b.c.A2)

Still worth to check that it works with zero (no `CustomOptionalClasses` option), one and two classes being passed.

Reply:
Added test for zero, one and two custom paths case.